docs/core_docs/docs/guides/langsmith_evaluation.md

```diff
@@ -0,0 +1,233 @@
+- Run regression tests on your application to confidently develop
+- Capture production analytics for product insights and continuous improvements
+
+## Prerequisites
+
+**[Create a LangSmith account](https://smith.langchain.com/) and create an API key (see bottom left corner). Familiarize yourself with the platform by looking through the [docs](https://docs.smith.langchain.com/)**
+
+Note LangSmith is in closed beta; we're in the process of rolling it out to more users. However, you can fill out the form on the website for expedited access.
+
+Now, let's get started!
+
+## Log runs to LangSmith
+
+First, configure your environment variables to tell LangChain to log traces. This is done by setting the `LANGCHAIN_TRACING_V2` environment variable to true.
+You can tell LangChain which project to log to by setting the `LANGCHAIN_PROJECT` environment variable (if this isn't set, runs will be logged to the `default` project). This will automatically create the project for you if it doesn't exist. You must also set the `LANGCHAIN_ENDPOINT` and `LANGCHAIN_API_KEY` environment variables.
+
+For more information on other ways to set up tracing, please reference the [LangSmith documentation](https://docs.smith.langchain.com/docs/).
+
+However, in this example, we will use environment variables.
+
+```bash
+npm install @langchain/core @langchain/openai langsmith uuid
+```
+
+```typescript
+import { v4 as uuidv4 } from "uuid";
+const unique_id = uuidv4().slice(0, 8);
+process.env.LANGCHAIN_TRACING_V2 = "true";
+process.env.LANGCHAIN_PROJECT = `JS Tracing Walkthrough - ${unique_id}`;
+process.env.LANGCHAIN_ENDPOINT = "https://api.smith.langchain.com";
+process.env.LANGCHAIN_API_KEY = "<YOUR-API-KEY>"; // Replace with your API key
+
+// For the chain in this tutorial
+process.env.OPENAI_API_KEY = "<YOUR-OPENAI-API-KEY>";
+// You can make an API key here: https://app.tavily.com/sign-in
+process.env.TAVILY_API_KEY = "<YOUR-TAVILY-API-KEY>";
+```
+
+Create the langsmith client to interact with the API
+
+```typescript
+import { Client } from "langsmith";
+
+const client = new Client();
+```
+
+Create a LangChain component and log runs to the platform. In this example, we will create a ReAct-style agent with access to a general search tool (DuckDuckGo). The agent's prompt can be viewed in the [Hub here](https://smith.langchain.com/hub/wfh/langsmith-agent-prompt).
+
+```typescript
+import { TavilySearchAPIRetriever } from "@langchain/community/retrievers/tavily_search_api";
+import { DocumentInterface } from "@langchain/core/documents";
+import { StringOutputParser } from "@langchain/core/output_parsers";
+import { ChatPromptTemplate } from "@langchain/core/prompts";
+import { RunnableLambda, RunnableMap } from "@langchain/core/runnables";
+import { ChatOpenAI } from "@langchain/openai";
+
+const retriever = new TavilySearchAPIRetriever({
+  k: 10,
+});
+const formatDocs = (docs: DocumentInterface[]) =>
```

Comment:
Might be better to use just `Document` for clarity

Reply:
Hm Document wasn't working with the type system but I'll see